### PR TITLE
Fixing the duplication bug in messaging.php

### PR DESCRIPTION
**Pull request by Knuvex**
- Fixed issue where message content was duplicated on page refresh

### CODE REVIEW

**Comment by tadhgboyle** [comment]
Thanks for the PR!
Are you able to do a screenrecording of the bug before and after the change?

**Comment by Derkades** [comment]
I will explain
- The browser does a POST request when submitting a form
- Nameless replies to that content with HTML
- The browser renders that HTML and shows it to the user. When the user refreshes the page, the browser has to do the previous request again, which in this case is a POST request. This is potentially unwanted, which is why browsers warn before doing this. However, users ignore warnings and the warning is ugly.

The fix is to do this instead:
- When receiving the POST request, instead of replying with HTML, reply with a redirect instead
- The browser does a second GET request, this time the proper HTML is returned
- Now, when refreshing, the browser will only reissue the second GET request which has no side-effects

File: modules/Core/pages/user/messaging.php [hunk 1]
```diff
@@ -406,6 +406,13 @@
                     // Display success message
                     $smarty->assign('MESSAGE_SENT', $language->get('user', 'message_sent_successfully'));
                     unset($_POST['content']);
+                    
+                    // Use strict comparison operator to compare request method
+                    if ($_SERVER['REQUEST_METHOD'] === 'POST') {
```
Comment (samerton) [inline modules/Core/pages/user/messaging.php:411]: There should be no need to do this check here as we'll already be within a POST request.

File: modules/Core/pages/user/messaging.php [hunk 2]
```diff
@@ -406,6 +406,13 @@
...
+                       // Redirect to same page to avoid resubmitting form on refresh
+                       header('Location: ' . $_SERVER['REQUEST_URI']);
```
Comment (samerton) [inline modules/Core/pages/user/messaging.php:413]: We can remove this line and the `exit;` afterwards and simply do the following:
```php
Redirect::to($_SERVER['REQUEST_URI']);
```

**Review by samerton — CHANGES_REQUESTED** [review]
Thanks for this! Just a couple of small things left as comments

**Pull request closed without merging**